avm/src/daemon/daemon.rs

```diff
@@ -26,29 +29,33 @@ use crate::vm::run::run;
 pub static CLUSTER_SECRET: OnceCell<Option<String>> = OnceCell::new();
 
 #[cfg(target_os = "linux")]
-async fn get_private_ip() -> String {
-  let res = Command::new("hostname").arg("-I").output().await;
-  let err = "Failed to execute `hostname`";
-  let stdout = res.expect(err).stdout;
-  String::from_utf8(stdout)
-    .expect(err)
-    .trim()
-    .split_whitespace()
-    .next()
-    .unwrap()
-    .to_string()
+async fn get_private_ip() -> Result<String, Box<dyn Error + Send + Sync>> {
```

Comment:
honestly, I think it makes sense to have a declaration like this that we just `use` everywhere to keep things DRY and reduce bloat
```rs
pub type Result<T> = std::result::Result<T, Box<dyn Error + Send + Sync>>;
```

Reply:
That is cleaner. It might need to be called something other than `Result` to avoid import conflicts though

Reply:
Nice one. I've called it `DaemonResult` to avoid collision with other results around.